vm/src/pyobject.rs

```diff
@@ -1044,150 +1232,58 @@ tuple_from_py_func_args!(A, B, C);
 tuple_from_py_func_args!(A, B, C, D);
 tuple_from_py_func_args!(A, B, C, D, E);
 
-impl<T> FromPyFuncArgs for T
-where
-    T: FromPyObject,
-{
-    fn required_params(ctx: &PyContext) -> Vec<Parameter> {
-        vec![Parameter {
-            kind: PositionalOnly,
-            typ: T::typ(ctx),
-        }]
-    }
-
-    fn from_py_func_args(args: &mut PyFuncArgs) -> PyResult<Self> {
-        Self::from_pyobject(args.shift())
-    }
-}
-
-pub type PyNativeFunc = Box<dyn Fn(&mut VirtualMachine, PyFuncArgs) -> PyResult>;
+pub type PyNativeFunc = Box<dyn Fn(&mut VirtualMachine, PyFuncArgs) -> PyResult + 'static>;
 
 pub trait PyNativeFuncFactory<T, R> {
```

Comment:
Slight dislike of the word factory here. A function has a trait for being a factory?

Is this actually IntoPyNativeFunc?

Reply:
Yeah, that's definitely a better name, changed